### PR TITLE
Use built-in "thermostat_mode" for HVAC mode

### PR DESCRIPTION
**Pull request by aivus**
Revert changes related to `hvac_mode` (added in #129) and use [capability options](https://apps.developer.homey.app/the-basics/devices/capabilities#enum-capability-options) for built-in `thermostat_mode`

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
-  "id": "com.gree",
-  "version": "0.8.0",
-  "compatibility": ">=5.0.0",
+  "id": "com.gree.test",
```
Comment (aivus) [inline app.json:2]: ⚠️ Test changes. Must be reverted.

File: app.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+  "version": "0.8.1",
+  "compatibility": ">=12.0.1",
   "sdk": 3,
   "brandColor": "#ff732e",
   "name": {
-    "en": "Gree"
+    "en": "Gree (test)"
```
Comment (aivus) [inline app.json:8]: ⚠️ Test changes. Must be reverted.

File: drivers/gree_cooper_hunter_hvac/device.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ class GreeHVACDevice extends Homey.Device {
         await this._executeCapabilityMigrations();
         this._registerCapabilityListeners();
 
-        this._markOffline();
+        // this._markOffline();
```
Comment (aivus) [inline drivers/gree_cooper_hunter_hvac/device.js:50]: ⚠️ Test changes. Must be reverted.

**Comment by aivus** [comment]
Hey @denniedegroot

I would be happy if you can install the version from this branch a do a quick testing. 
Unfortunately, I don't have access to my AC and cannot test the changes with the real device.

With this changes I'm returning back to `thermostat_mode` instead of `hvac_mode` because now we can use [capability options](https://apps.developer.homey.app/the-basics/devices/capabilities#enum-capability-options) to pass custom enum values.

Btw, this article [might be interesting](https://apps.developer.homey.app/upgrade-guides/device-capabilities) because Athom is going to "fix" firmware of Homey Pro 2023 and sync the behaviour with old Homey.

**Comment by denniedegroot** [comment]
@aivus I did a quick test. Everything seems to work as intended, the only thing I noticed was that the initial on/off state was incorrect.

I noticed a bug I introduced recently, a patch is below:
```
diff --git a/app.js b/app.js
index 5ec5d5a..dff0216 100644
--- a/app.js
+++ b/app.js
@@ -62,7 +62,7 @@ class GreeHVAC extends Homey.App {
 
         this.homey.flow.getConditionCard('quiet_mode_is')
             .registerRunListener((args, state) => {
-                const quietMode = args.device.getCapabilityValue('quiet_mode_is');
+                const quietMode = args.device.getCapabilityValue('quiet_mode');
                 args.device.log('[condition]', '[quiet mode]', quietMode);
                 return args.mode === quietMode;
             });

```

**Comment by aivus** [comment]
Thanks for the quick response @denniedegroot
I wasn't sure about the flow changes (triggers/conditions/actions) as it requires real interaction with the device I don't currently have.

I'll push the changes to the `test` channel today.

> the only thing I noticed was that the initial on/off state was incorrect.

I was noticing this for several years already when initial state of UI for on/off is not in sync with value in the capability. I was contacting Athom several years ago and it wasn't fixed.

> I noticed a bug I introduced recently, a patch is below:

Thanks. I'll apply the patch.

**Pull request merged**